src/bin/ion/commands/generate/generator.rs

```diff
@@ -365,7 +361,20 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
             .push(data_model_node.to_owned());
 
         // pop out the nested type name from the fully qualified namespace as it has been already added to the type store and to nested types
-        self.current_type_fully_qualified_name.pop();
+        // For sequence type, it would already have popped out the nested type name.
+        if !data_model_node.is_sequence() {
```

Comment:
(🗺️ PR tour) Since nested sequence already removes resets its namespace in `build_sequence_from_constraints`. This is not required for nested sequence.

Reply:
Why not just give each `AbstractDataType` its own copy of it's fully qualified name as a `Vec<String>`?

Reply:
Shifted this logic to `convert_isl_constraints_to_data_model()`, now has a single place where namespace will be reset.